### DISCUSSION TITLE
Does Jobrunr server resume scheduled jobs after server downtime or restart

### DISCUSSION THREAD

**Discussion by omertopuz** [jobrunr/jobrunr#706] [General]
I scanned the documentation but I couldn’t find anything about this topic. My concern is to resume the scheduled jobs for an instant. Here is what;
I setup Jobrunr to schedule jobs at its given time. Let' s say, there are 3000 jobs scheduled at different instants in the future. At some point, the server crashed or stopped manually for any reason. 
Q1: Are all jobs resumed again and executed at their given times
Q2: Let' s say, There are 200 jobs in this 3000 jobs of which the scheduled time is over during the server downtime. Then I restarted the server. is this 200 jobs executed even though they are over

**Comment by rdehuyss** [comment]
Hi @omertopuz, here some answers:
Q1: all scheduled jobs will run again at their scheduled time as soon as the server is up again. All scheduled jobs that have a scheduled time in the past (because they were not executed during downtime), will run as soon as the server is started.

Q2: I think this is answered by my answer above. If not, please rephrase your question.

**Reply by rdehuyss** [reply]
Note that the above is for scheduled jobs. With recurring jobs, this is different and explained in the docs.

**Reply by omertopuz** [reply]
Hi @rdehuyss , thank you for your quick response. I have one more question. Assume I have a scheduled job in `Enqueued` status and the job hasn’t been finished yet. The server is down or the job takes much time to finish in the mean time. when the server restarts does the job executed again? Does the job go from `Enqueued` to `Succeded`

**Reply by rdehuyss** [reply]
Yes, as long as it did not succeed completely, it will be restarted.

**Reply by rdehuyss** [reply]
Feel free to star the repo if you're happy with the quick answers 😉

**Reply by omertopuz** [reply]
Thank you. I starred. It was really helpful for me.

**Reply by omertopuz** [reply]
Hi @rdehuyss 
it is me again but with a different question related to this topic. I verified the cases mentioned above and the scheduled jobs start after the server downtime. It is OK. But the scheduled jobs are starting 1-2 minutes later the server is up, I mean they are not triggered immediately. Is there any way of configuring this behaviour? I mean, I would like to send a parameter which indicates the delay to trigger the jobs at the very beginning of server is up and running. Because I need to prepare some data to execute the jobs and I am planning to finish the preparation in this delay duration.
Thanks in advance

**Reply by rdehuyss** [reply]
Hi @omertopuz , there are ways to do so but only in the Pro version I'm afraid. I need to feed my kids too 😉 .

**Reply by omertopuz** [reply]
Ok. thanks for the reply. we will discuss upgrading to Pro version in the team

**Reply by omertopuz** [reply]
Hi @rdehuyss 
we have one more question to decide on Jobrunr in our project. Suppose that, I submitted 1000 scheduled jobs and 300 are already triggered on time. In the mean time, the server crashed for a while. But these 300 jobs are not done with their inner processing yet. When the server restart, these jobs should be triggered again. When I look at the Jobrunr scheduled jobs, they only trigger the method (Job lamda) and goes into `SUCCEDED` state. So the question is,

Is there any structure which waits a signal from JobLamda and goes into `SUCCEDED` state. This will enable to keep the job in `PROCESSING` state and once the server restart these jobs will triggered again since they are already in `PROCESSING` state based on your previous explanations. I need to somehow keep job state in `PROCESSING`. is it doable

**Reply by rdehuyss** [reply]
Yes, as long as the job does not succeed, it will be restarted. These jobs will go to a failed state first as they are orphaned and then they will be picked up again.
